lua/clear-action/popup.lua

```diff
@@ -28,11 +28,23 @@ M.hide_cursor_autocmd = function()
   })
 end
 
+local function client_name(client_id)
+  local client = vim.lsp.get_client_by_id(client_id)
+  if client then
+    return " " .. client.name
+  else
+    return ""
+  end
```

Comment:
You could make this in one line, and add the space only when concatenating with the row

```suggestion
  return vim.lsp.get_client_by_id(client_id) or ""
```